cmake/app/boilerplate.cmake

```diff
@@ -193,6 +193,12 @@ endif()
 assert(BOARD "BOARD not set")
 message(STATUS "Board: ${BOARD}")
 
+include(${ZEPHYR_BASE}/board_aliases.cmake OPTIONAL)
```

Comment:
Wouldn't this be better placed in `boards/`?

Reply:
It could be.  I prefer to keep stuff that isn't under configuration control (e.g. `hw.yml` for hardware sanitycheck testing) in the top level directory, where it's a little easier to preserve from `git clean -fdx subdir` . I'd actually prefer this to be stored out of tree somewhere, but I don't know how to do that in a way that CMake can find it.

Reply:
I understand the need and welcome this enhancement. However I do 'git clean -dxf' on the whole tree all the time, so this wont work, we need to keep this out of the tree to make it usefull. Look at how we store caches in user directory in cmake/extensions.cmake

Reply:
@nashif `${USER_CACHE_DIR}` would be a bad location since it's undocumented and this file is expected to be maintained by the user, who may know nothing about that directory or its value.

Other options:
* create a new variable `ZEPHYR_OOT_DIR` or something less horrible?
* put it in what passes for ${HOME} and name it `.zephyr_board_aliases.cmake`
* ???

I don't like any of these alternatives, so if we can't come up with a place to put managed Zephyr customization files out of tree my next approach is to support it in west.

Reply:
> `${USER_CACHE_DIR}` would be a bad location since it's undocumented and this file is expected to be maintained by the user, who may know nothing about that directory or its value.

was not suggesting to use the cache dir, just use the same way we write out of the tree.